### PR TITLE
aioredis client with buffer queue

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: redisgraph_bulk_loader/bulk_insert.py [hunk 1]
```diff
@@ -70,7 +70,8 @@ def process_entities(entities):
 @click.option('--max-token-size', '-t', default=500, help='max size of each token in megabytes (default 500, max 512)')
 @click.option('--index', '-i', multiple=True, help='Label:Propery on which to create an index')
 @click.option('--full-text-index', '-f', multiple=True, help='Label:Propery on which to create an full text search index')
-def bulk_insert(graph, host, port, password, user, unix_socket_path, nodes, nodes_with_label, relations, relations_with_type, separator, enforce_schema, skip_invalid_nodes, skip_invalid_edges, escapechar, quote, max_token_count, max_buffer_size, max_token_size, index, full_text_index):
+@click.option('--async-requests', '-A', default=3, help='amount of async requests to be executed in parallel' )
```
Comment (jeffreylovitz) [inline redisgraph_bulk_loader/bulk_insert.py:73]: `number` instead of `amount`

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,6 @@
-click>=7.0
+asyncclick
+click
+anyio
```
Comment (jeffreylovitz) [inline requirements.txt:3]: Are these actually dependencies?

File: redisgraph_bulk_loader/bulk_insert.py [hunk 3]
```diff
@@ -70,7 +70,8 @@ def process_entities(entities):
...
+async def bulk_insert(graph, host, port, password, user, unix_socket_path, nodes, nodes_with_label, relations, relations_with_type, separator, enforce_schema, skip_invalid_nodes, skip_invalid_edges, escapechar, quote, max_token_count, max_buffer_size, max_token_size, index, full_text_index, async_requests):
```
Comment (jeffreylovitz) [inline redisgraph_bulk_loader/bulk_insert.py:74]: If I understand correctly, this function is `async` because otherwise it cannot make `async` calls?

**Review by jeffreylovitz — CHANGES_REQUESTED** [review]
This looks good to me, a few comments!

**Comment by swilly22** [comment]
@DvirDukhan please move `async_requests` into config.py, remove it from:
`query_buf = QueryBuffer(graph, client, config, async_requests)`

File: requirements.txt (hunk 2 above)
Comment (DvirDukhan) [reply requirements.txt:3]: will be removed after rebase

File: redisgraph_bulk_loader/bulk_insert.py (hunk 3 above)
Comment (DvirDukhan) [reply redisgraph_bulk_loader/bulk_insert.py:74]: correct

**Pull request closed without merging**